sn_networking/src/event.rs

```diff
@@ -126,8 +126,8 @@ pub enum NetworkEvent {
     PeerAdded(PeerId),
     // Peer has been removed from the Routing Table
     PeerRemoved(PeerId),
-    /// The Records for the these keys are to be fetched from the network
-    KeysForReplication(Vec<RecordKey>),
+    /// The records bearing these keys are to be fetched from the holder or the network
+    KeysForReplication(Vec<(PeerId, RecordKey)>),
```

Comment:
@maqi might we not want a vec of peerids here? In case we already fetch a record from one peer?

Reply:
This is for the notification from swarm_driver to node to carry out the `fectch` action, which has to have a `PeerId` info bundled with `RecordKey` to tell from where to fetch.

Regarding the `parallel fetch from one peer`, there is `MAX_PARALLEL_FETCH` (currently set to 16) within the replication_fetcher to avoid choking a node by sending too many fetch request to it.

Regarding make the replication_fetcher avoid duplicate fetching same record from different peers, I just added a filter at https://github.com/maqi/stableset_net/blob/further_investigate/sn_networking/src/replication_fetcher.rs#L93
let me know if it make sense.